app/src/main/java/com/courseproject/tindar/ChatRecyclerViewAdapter.java

```diff
@@ -0,0 +1,145 @@
+
+    /**
+     * Constructor. ChatRecyclerViewAdapter is intended to be an adapter for a view.
+     * @param displayedMessages list of messages that this adapter will help display
+     * @param userID User ID of the current user using the app
+     */
+    public ChatRecyclerViewAdapter(ArrayList<MessageModel> displayedMessages,
+                                   String userID){
+        this._displayedMessages = displayedMessages;
+        this._userID = userID;
+    }
+
+    /**
+     * Creates and adds TindarMessageViewHolder objects to the parent ViewGroup
+     * @param parent   The ViewGroup into which the new View will be added after it is bound to
+     *                 an adapter position.
+     * @param viewType The view type of the new View.
+     * @return
+     */
+    @NonNull
+    @Override
+    public TindarMessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        /*
+        this is where we will inflate the layout;
+        take the layout description from the XML
+        and populate the view hierarchy with actual View objects.
+        TODO: refactor to avoid switch statements
+         */
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View view;
+        switch(viewType){
+            case 0:
+                view = inflater.inflate(R.layout.recycler_row_message_sent,
+                        parent, false);
+                return new TindarMessageViewHolder(view);
+            case 1:
+                view = inflater.inflate(R.layout.recycler_row_message_received,
+                        parent, false);
+                return new TindarMessageViewHolder(view);
+        }
+        return null;
+    }
+
+    /**
+     * Updates a message TindarMessageViewHolder with the relevant details
+     * from the corresponding MessageModel
+     * @param holder   The ViewHolder which should be updated to represent the contents of the
+     *                 item at the given position in the data set.
+     * @param position The position of the item within the adapter's data set.
+     */
+    @Override
+    public void onBindViewHolder(@NonNull TindarMessageViewHolder holder, int position) {
+        // TODO: refactor for easier future support of other message types
+        holder.messageContentLayout.setText(_displayedMessages.get(position).getMessageContent());
+    }
+
+    /**
+     * @return number of items within the adapter's data set.
+     */
+    @Override
+    public int getItemCount() {
+        // I think this is a helper for onBindViewHolder()?
+        return (this._displayedMessages).size();
+    }
+
+    // TODO: refactor
+    /**
+     * Categorizes items being cycled through
+     * Feeds into onCreateViewHolder.
+     * @param position
+     * @return 0 if the message originated from the current user; 1 otherwise
+     */
+    @Override
+    public int getItemViewType(int position){
```

Comment:
why return type is not boolean if it's just 0 and 1?

Reply:
implementing an interface I didn't define; can't do that.